subsys/bluetooth/host/id.c

```diff
@@ -363,6 +363,11 @@ int bt_id_set_private_addr(uint8_t id)
 		return -EINVAL;
 	}
 
+	/* bt_rand is not available before Bluetooth enable has been called */
+	if (!atomic_test_bit(bt_dev.flags, BT_DEV_ENABLE)) {
+		return -EAGAIN;
+	}
```

Comment:
Hmm, since `bt_rand` is actually a public function, should we actually move the check for `BT_DEV_ENABLE` into `bt_rand` instead?

Reply:
I initially thought about that, but I found the same check used somewhere in the file, so I just followed.
Also, I'm not sure if `bt_rand` knows about `BT_DEV_ENABLE` flag.

Reply:
Ok, I dropped that change and created a separate PR for `bt_rand()`